frontend/src/metabase/visualizations/visualizations/CartesianChart/events.ts

```diff
@@ -178,3 +194,149 @@ export const canBrush = (
     !hasClickBehavior(series)
   );
 };
+
+export const getSeriesHoverData = (
```

Comment:
I extracted these get-data-for-event functions from the component itself to make it more readable